### PR TITLE
reverted the efs-utils version to v1.33.4 pre-release v1.4.4

### PR DESCRIPTION
**Pull request by mskanth972**
**Is this a bug fix or adding new feature?**

**What is this PR about? / Why do we need it?**

**What testing is done?**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -40,7 +40,7 @@ RUN if [ "$EFSUTILSSOURCE" = "yum" ]; \
          yum -y install git rpm-build make && \
          git clone https://github.com/aws/efs-utils && \
          cd efs-utils && \
-         git checkout $(git describe --tags $(git rev-list --tags --max-count=1)) && \
+         git checkout $(git describe --tags 71058a0fed32311e67403e18ca715d0283f94282) && \
```
Comment (Ashley-wenyizha) [inline Dockerfile:43]: How did we test this?

File: CHANGELOG-1.x.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# V1.4.4
+*Post release update to v1.4 branch (v1.4.3) ([#810](https://github.com/kubernetes-sigs/aws-efs-csi-driver/pull/810), [@Ashley-wenyizha](https://github.com/Ashley-wenyizha))
```
Comment (Ashley-wenyizha) [inline CHANGELOG-1.x.md:2]: Nit : trivial change and not worthy to mention in the release, can remove

Not Nit : Add efs-utils version change in this new release and brief mention why the release of this specific version

**Review by Ashley-wenyizha — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (mskanth972) [reply Dockerfile:43]: git describe --tags 71058a0fed32311e67403e18ca715d0283f94282
v1.33.4

**Review by Ashley-wenyizha — COMMENTED** [review]
lgtm, waiting for e2e to finish

**Pull request merged**